media/server/gstplayer/source/GstGenericPlayer.cpp

```diff
@@ -387,10 +387,11 @@ bool GstGenericPlayer::getPosition(std::int64_t &position)
     return true;
 }
 
-GstElement *GstGenericPlayer::getSink(const MediaSourceType &mediaSourceType)
+GstElement *GstGenericPlayer::getSink(GstObject *&pointerToUnref, const MediaSourceType &mediaSourceType) const
```

Comment:
How about instead of adding this complexity, we make it so the returned sink always has to be unrefed.

We could do this by increasing the reference of autosink (if there is one) using gst_object_ref in getSinkChildIfAutoVideoSink.